# p53-chip-seq-go

Generate GO results table/graphs for P53 genomics project

run script: `./run.sh` (see "Building" section)

## About

The purpose of this script is to determine highly enriched GO terms

Originally, terms were filtered to those which matched keywords such as 'cell death', 'apoptosis', and 'caspase'. However, filtering based on the keywords did not 

## Data Source

GO analysis files generated from Homer analysis of P53 ChIP-Seq files

## Process

For each folder in the DmGOs directory, there is a `geneOntology.html` file.

For each `geneOntology.html` file, top 10 rows with lowest p-value are outputted to a results file with the term, p-value, and level 4 ancestor traceback. The level 4 ancestor traceback may contain multiple terms, including the original term if it is level 4. Level 4 is used because higher levels (0~3) are too generic.

The level 4 traceback is based on the Gene Ontology data file from [http://geneontology.org/page/download-ontology]().


## Requirements

1. beautifulsoup4
2. goatools
  * fisher
    * numpy

## Directory tree

The following directory tree is assumed/generated by `P53_ChIPSeq_GO.py`:

```
BASE_DIR
│
├───p53-chip-seq-go (this Git repository)
│   │   P53_ChIPSeq_GO.py
│   │   run.sh
│   │   ...
│   │
│
│
├───data
│   ├───go-basic.obo
│   └───DmGOs
│       ├───Kc167_P53_NT60_A_GO
│       │   │   geneOntology.html
│       │   │   ...
│       │
│       ├───Kc167_P53_NT60_B_GO
│       │   ...
│
│
└───results
    └───P53-ChIPSeq-GO-results
        ├───Kc167_P53_NT60_A_GO-results.txt
        ├───Kc167_P53_NT60_B_GO-results.txt
        │   ...
        │
```

## Building

Make sure you are in the `p53-chip-seq-go` directory.

Run `P53_ChIPSeq_GO.py` with `python` and `virtualenv`:

```
./run.sh
```

The repository comes with a virtual environment (`ve/`). If `run.sh` does not work, try the teardown and bootstrap process:

```
./teardown.sh
```

```
./bootstrap.sh
```
